### ISSUE TITLE
`pl.*_horizontal()` exprs are not JSON-serializable except any/all

### ISSUE THREAD

**Issue by aldanor** [pola-rs/polars#11532]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
pl.sum_horizontal(pl.all()).meta.write_json()
```

### Log output

```shell
ValueError: Error("serialize not supported for this 'opaque' function", line: 0, column: 0)
```

### Issue description

All `pl.*_horizontal()` functions _except_ `all_horizontal` / `any_horizontal` don't seem to be json-serializable.

### Expected behavior

-

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.3
Index type:          UInt32
Platform:            macOS-13.4.1-arm64-arm-64bit
Python:              3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:33:12) [Clang 15.0.7 ]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
gevent:              <not installed>
matplotlib:          3.8.0
numpy:               1.24.4
pandas:              2.1.1
pyarrow:             13.0.0
pydantic:            <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

**Comment by ritchie46** [comment]
Yes, the `fold` currently uses the `Anonymous` function. These should be rewritten to our `Expr::Function`  architecture and then it will inherit all the good stuff like serialization and optimizations.

**Comment by romanovacca** [comment]
In #10412 one can see how the rewriting to `Expr::Function` architecture was done for `all_horizontal / any_horizontal`

**Comment by romanovacca** [comment]
The Expr::Function architecture functionality was implemented in #11685 and #11659

running the MRE now gives:
```
>>> pl.sum_horizontal(pl.all()).meta.write_json()
'{"Function":{"input":["Wildcard"],"function":"SumHorizontal","options":{"collect_groups":"ApplyFlat","fmt_str":"","input_wildcard_expansion":true,"auto_explode":true,"cast_to_supertypes":false,"allow_rename":true,"pass_name_to_apply":false,"changes_length":false,"check_lengths":true,"allow_group_aware":true}}}'
```

@stinodego this can be closed?
